src/gui/entry/EntryView.cpp

```diff
@@ -144,24 +268,262 @@ Entry* EntryView::entryFromIndex(const QModelIndex& index)
+    m_inSearchMode = false;
 }
 
-void EntryView::switchToGroupMode()
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Switch to search mode, i.e. list search results (NOTE: previously named
+ * 'switchToEntryListMode')
+ */
+void EntryView::switchToSearchMode()
 {
-    m_sortModel->hideColumn(0, true);
+    /* Check if already in this mode */
+    if (m_inSearchMode) {
+        return;
+    }
+
+    /*
+     * Use header()->showSection() instead of m_sortModel->hideColumn() as
+     * the latter messes up column indices, interfering with code relying on
+     * proper indices
+     */
+    header()->showSection(EntryModel::ParentGroup);
 
-    m_sortModel->sort(-1, Qt::AscendingOrder);
-    m_sortModel->sort(0, Qt::AscendingOrder);
-    sortByColumn(0, Qt::AscendingOrder);
+    /*
+     * Always set sorting to column 'Group', as it does not feel right to have
+     * the last known sort configuration of search view restored by 'Database
+     * WidgetStateSync', which is what happens without this
+     */
+    m_sortModel->sort(EntryModel::ParentGroup, Qt::AscendingOrder);
+    sortByColumn(EntryModel::ParentGroup, Qt::AscendingOrder);
+
+    m_inSearchMode = true;
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Get current state of 'Hide Usernames' setting (NOTE: just pass-through for
+ * m_model)
+ */
+bool EntryView::hideUsernames() const
+{
+    return m_model->hideUsernames();
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Set state of 'Hide Usernames' setting (NOTE: just pass-through for m_model)
+ */
+void EntryView::setHideUsernames(const bool hide)
+{
+    m_model->setHideUsernames(hide);
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Get current state of 'Hide Passwords' setting (NOTE: just pass-through for
+ * m_model)
+ */
+bool EntryView::hidePasswords() const
+{
+    return m_model->hidePasswords();
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Set state of 'Hide Passwords' setting (NOTE: just pass-through for m_model)
+ */
+void EntryView::setHidePasswords(const bool hide)
+{
+    m_model->setHidePasswords(hide);
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Get current state of view
+ */
+QByteArray EntryView::viewState() const
+{
+    return header()->saveState();
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Set state of entry view view
+ */
+bool EntryView::setViewState(const QByteArray& state) const
+{
+    return header()->restoreState(state);
+}
+
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Sync checkable menu actions to current state and display header context
+ * menu at specified position
+ */
+void EntryView::showHeaderMenu(const QPoint& position)
+{
+    /* Sync checked state of menu actions to current state of view */
+    m_hideUsernamesAction->setChecked(m_model->hideUsernames());
+    m_hidePasswordsAction->setChecked(m_model->hidePasswords());
+    foreach (QAction *action, m_columnActions->actions()) {
```

Comment:
Use C++11 range-based for loop (make sure to use asConst to cast the container to const to avoid detaching).

Reply:
@fonic good article about [Qt containers and C++11 range-based loops](https://www.dvratil.cz/2015/06/qt-containers-and-c11-range-based-loops/)